src/main/java/de/tum/in/www1/artemis/service/ComplaintResponseService.java

```diff
@@ -160,69 +160,104 @@ public ComplaintResponse createComplaintResponseRepresentingLock(Complaint compl
+     * This method handles updating the state of a complaint based on user-provided acceptance and response text.
+     *
+     * @param updatedComplaintResponse The DTO containing updated response details such as acceptance state and response text.
+     * @param complaintResponseId      The ID of the complaint response to be updated.
+     * @return ComplaintResponse The updated and persisted complaint response reflecting the new state and information.
+     */
+    public ComplaintResponse resolveComplaint(ComplaintResponseUpdateDTO updatedComplaintResponse, Long complaintResponseId) {
+        return processComplaint(complaintResponseId, updatedComplaintResponse, updatedComplaintResponse.complaintIsAccepted(), updatedComplaintResponse.responseText());
+    }
+
+    /**
+     * Resolves a complaint using an updated ComplaintResponse object. This method processes the complaint
+     * based on the inherent acceptance status and response text within the updated ComplaintResponse.
      *
-     * @param updatedComplaintResponse complaint response containing the information necessary for resolving the complaint
-     * @return complaintResponse of resolved complaint
+     * @param updatedComplaintResponse The updated ComplaintResponse object containing the new state, such as acceptance and response text.
+     * @return ComplaintResponse The updated and persisted complaint response reflecting the changes made.
      */
     public ComplaintResponse resolveComplaint(ComplaintResponse updatedComplaintResponse) {
-        if (updatedComplaintResponse.getId() == null) {
-            throw new IllegalArgumentException("The complaint response needs to have an id");
+        return processComplaint(updatedComplaintResponse.getId(), null, updatedComplaintResponse.getComplaint().isAccepted(), updatedComplaintResponse.getResponseText());
+    }
+
+    /**
+     * Processes the updating of a complaint response. This method consolidates the common logic for updating complaint responses,
+     * including validating user permissions, checking lock status, validating complaint details, and updating the complaint
+     * in the database.
+     *
+     * @param complaintResponseId      The ID of the complaint response to be updated.
+     * @param updatedComplaintResponse Optional DTO containing additional complaint response updates, null if not applicable.
+     * @param isAccepted               Boolean indicating whether the complaint has been accepted or rejected.
+     * @param responseText             String containing the response text associated with the complaint response.
+     * @return ComplaintResponse The updated and persisted complaint response.
+     * @throws IllegalArgumentException if the acceptance status is not clearly defined (null).
+     */
+    private ComplaintResponse processComplaint(Long complaintResponseId, ComplaintResponseUpdateDTO updatedComplaintResponse, Boolean isAccepted, String responseText) {
```

Comment:
`updatedComplaintResponse` is never used in the method.

```suggestion
    private ComplaintResponse processComplaint(Long complaintResponseId, Boolean isAccepted, String responseText) {
```